### PR TITLE
Fallback to sms if target is not a valid whatsapp user

### PR DESCRIPTION
**Pull request by tung2744**
ref #3528

### CODE REVIEW

**Comment by tung2744** [comment]
Updated to return an api error when whatsapp user is not available.

**Review by louischan-oursky — COMMENTED** [review]
1. When `phone_otp_mode` is `whatsapp`, we cannot fallback, the error message is unknown error. We should recognize the error and tell the number is not a Whatsapp number. 

https://github.com/authgear/authgear-server/assets/15082564/accca584-2cd2-48da-a09a-4cb289a172fb

2. When `phone_otp_mode` is `whatsapp_sms`, we can fallback. After fallback, we can see the branch of switching back to whatsapp (I think it is fine to show this). Clicking the branch will reload the page and make the branch disappear. I think we show error in this case. That is, when the branch is explicitly taken by the end-user, we do not fallback and show error instead.


https://github.com/authgear/authgear-server/assets/15082564/d6f23046-2d7c-4c0c-81a0-fb930bd3a691

**Comment by tung2744** [comment]
@louischan-oursky Updated:

1. When user explicitly selected whatsapp, an error will be shown instead of fallback to sms.
https://github.com/authgear/authgear-server/assets/34942210/d29733b5-e687-4955-b27f-9384f0deedf2

2. When the system selected whatsapp automatically and found it is not a valid whatsapp user, fallback to sms and the whatsapp branch will be hidden.
https://github.com/authgear/authgear-server/assets/34942210/06a8b8b9-72a1-4d01-a193-aac563e6b7c1

**Comment by tung2744** [comment]
@louischan-oursky Updated again to always show all options

File: pkg/lib/authn/otp/errors.go [hunk 1]
```diff
@@ -12,7 +12,7 @@ var ErrCodeNotFound = InvalidOTPCode.NewWithCause("otp code is expired or invali
 var ErrInvalidCode = InvalidOTPCode.NewWithCause("invalid otp code", apierrors.StringCause("InvalidCode"))
 var ErrConsumedCode = InvalidOTPCode.NewWithCause("used otp code", apierrors.StringCause("UsedCode"))
 
-var ErrInvalidWhatsappUser = InvalidOTPCode.NewWithCause("invalid whatsapp user", apierrors.StringCause("InvalidWhatsappUser"))
+var ErrInvalidWhatsappUser = InvalidWhatsappUser.NewWithCause("invalid whatsapp user", apierrors.StringCause("InvalidWhatsappUser"))
```
Comment (louischan-oursky) [inline pkg/lib/authn/otp/errors.go:15]: No need to specify cause here. Just call `New("invalid whatsapp user")`

Comment (tung2744) [reply pkg/lib/authn/otp/errors.go]: @louischan-oursky Updated, thanks!

**Pull request merged**